lib/archethic_web/controllers/api/transaction_controller.ex

```diff
@@ -25,21 +38,27 @@ defmodule ArchethicWeb.API.TransactionController do
           |> TransactionPayload.to_map()
           |> Transaction.cast()
 
-        case Archethic.send_new_transaction(tx) do
-          :ok ->
-            TransactionSubscriber.register(tx.address, System.monotonic_time())
-
-            conn
-            |> put_status(201)
-            |> json(%{
-              transaction_address: Base.encode16(tx.address),
-              status: "pending"
-            })
-
-          {:error, :network_issue} ->
-            conn
-            |> put_status(422)
-            |> json(%{status: "error - transaction may be invalid"})
+        tx_address = tx.address
+
+        storage_nodes = Election.chain_storage_nodes(tx_address, P2P.available_nodes())
```

Comment:
I think you should use authorized and available nodes to make it consistent with the shard election.

Reply:
Yes that would be better